File: src/main/kotlin/de/brainsizzle/twitchbotpaint/paint/CirclePainter.java
package de.brainsizzle.twitchbotpaint.paint;

import java.awt.*;

public class CirclePainter {

    public void drawCircle(Graphics2D g2, Shape shape) {
        g2.setColor(shape.getRenderingColor());
        g2.setStroke(new BasicStroke(3.0f));
        g2.drawOval(
                calcCenter(shape.getPosition().getXCoordinate(), shape.getSize()),
                calcCenter(shape.getPosition().getYCoordinate(), shape.getSize()),
                shape.getSize(), shape.getSize()
        );
    }

    public int calcCenter(int coordinate, int size) {
        return coordinate - size / 2;
    }

}

File: src/main/kotlin/de/brainsizzle/twitchbotpaint/paint/CirclePainter.kt
package de.brainsizzle.twitchbotpaint.paint

import java.awt.BasicStroke
import java.awt.Graphics2D

class CirclePainter {

    fun drawCircle(g2: Graphics2D, shape: Shape) {
        g2.color = shape.getRenderingColor()
        g2.stroke = BasicStroke(3.0f)

        if (shape.fill) {
            g2.fillOval(
                shape.calcBoundingBoxLeft(),
                shape.calcBoundingBoxTop(),
                shape.size, shape.size
            )
        } else {
            g2.drawOval(
                shape.calcBoundingBoxLeft(),
                shape.calcBoundingBoxTop(),
                shape.size, shape.size
            )
        }
    }
}